### ISSUE TITLE
[require-param] Add `contexts`-option

### ISSUE THREAD

**Issue by manuth** [gajus/eslint-plugin-jsdoc#511]
I'm working with TypeScript and I'd like to require `@param`-tags in interface-methods (context: `TSMethodSignature`).

```ts
interface ITest {
    /**
     * Test description.
     */
    TestMethod(id: number): void;
}
```

Sadly the `require-param` rule doesn't allow me to specify a custom `contexts`-array.

**Comment by manuth** [comment]
@brettz9 wow - that was fast!
Thank you so much for taking care 😄

**Comment by brettz9** [comment]
Thanks for the idea and well thought-out suggestions! Also happened to be pretty easy to add given our existing infrastructure for handling `contexts`/`exemptedBy`...

**Comment by manuth** [comment]
@brettz9 since updating `eslint-plugin-json` to the most recent version I get an error when enabling `require-param` for `MethodDefinition`s:

```
Cannot read property 'map' of undefined\nOccurred while linting test.js:9
```

Is there some way I can debug the rule?

**Comment by brettz9** [comment]
You could go into `eslint-plugin-json` source and log (and that would be helpful so the error can at least be cleared up for others), particularly if running eslint from the command line.

But I'm not actually sure there is a use case for adding it on `MethodDefinition` since  the AST seems it will have `FunctionExpression`, at least for regular `class` methods.

**Comment by manuth** [comment]
Thank you very much for the info - running it from command line actually helped me very much!
I had an incorrect context in the array chich caused that at this piece of code:
https://github.com/gajus/eslint-plugin-jsdoc/blob/bd432f68bb57d91dd49a9ddc1fbaf69ff7aa4678/src/jsdocUtils.js#L36-L37
`functionNode.params` was undefined.

Sorry for the confusion - looks like I ducked up my eslint-configuration :')

**Comment by brettz9** [comment]
No worries. if you find a legitimate context of course we can change the code.
